### PR TITLE
renamed from (deprecated) service to kleisli

### PR DESCRIPTION
**Pull request by TomTriple**

### CODE REVIEW

File: server/src/main/scala/org/http4s/server/syntax/package.scala [hunk 1]
```diff
@@ -7,11 +7,11 @@ import cats.implicits._
 
 package object syntax {
   @deprecated(
-    "Import `cats.implicits._` and use `service1 <+> service2` instead. Ensure -Ypartial-unification is enabled.",
+    "Import `cats.implicits._` and use `kleisli1 <+> kleisli` instead. Ensure -Ypartial-unification is enabled.",
```
Comment (diesalbla) [inline server/src/main/scala/org/http4s/server/syntax/package.scala]:
```suggestion
    "Import `cats.implicits._` and use `kleisli1 <+> kleisli2` instead. Ensure -Ypartial-unification is enabled.",
```

File: core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala [hunk 2]
```diff
@@ -8,8 +8,8 @@ import cats.data.{Kleisli, OptionT}
 
 trait KleisliSyntax {
   implicit def http4sKleisliResponseSyntax[F[_]: Functor, A](
-      service: Kleisli[OptionT[F, ?], A, Response[F]]): KleisliResponseOps[F, A] =
-    new KleisliResponseOps[F, A](service)
+      kleisli: Kleisli[OptionT[F, ?], A, Response[F]]): KleisliResponseOps[F, A] =
```
[the remark this answers is not in the document]
Comment (TomTriple) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: Interesting to know @rossabaker ! I did not think for a second about that.

File: core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala [hunk 3]
```diff
@@ -8,8 +8,8 @@ import cats.data.{Kleisli, OptionT}
 
 trait KleisliSyntax {
   implicit def http4sKleisliResponseSyntax[F[_]: Functor, A](
```
[the remark this answers is not in the document]
Comment (TomTriple) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: Renaming breaks a test in BlazeClientSpec, so I removed that again ;)

[the remark this answers is not in the document]
Comment (rossabaker) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: `BlackClientSpec` is regrettably flaky, so unless it broke locally, I'll bet it wasn't you who broke it.

File: core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (rossabaker) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: This is something to be more careful of on things like case classes, or functions and constructors that take a few arguments.  Particularly when functions take more than one argument of the same type, people are likely to call things with named parameters.  But yeah, it's rare for people to call by name on unary functions, and even rarer on implicits.

File: core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomTriple) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: I'll have a look at it again when I'm back home

[the remark this answers is not in the document]
Comment (TomTriple) [reply core/src/main/scala/org/http4s/syntax/KleisliSyntax.scala]: You won your bet, it worked locally!

**Pull request merged**